pkg/controller/registry/reconciler/grpc.go

```diff
@@ -168,6 +169,78 @@ func (c *GrpcRegistryReconciler) ensureService(source grpcCatalogSourceDecorator
 	return err
 }
 
+// updateRegistryPodByDigest is an internal method that verifies the latest catalog source by comparing image digests.
+// If the digests do not match (i.e. there is a new version of the catalog source) then overwrite the catalog source pod.
+// This is done to only have the container runtime (CRI-O) talk to the container registry.
+func (c *GrpcRegistryReconciler) updateRegistryPodByDigest(source grpcCatalogSourceDecorator) bool {
+	podDigestChan := make(chan string, 1)
+	lastCheckedTimestamp := time.Time{}
+	interval := source.Spec.Poll.Interval.Duration
+
+	// check poll value is not zero (default poll value)
+	// if polling interval is zero polling will not be done
+	if interval == time.Duration(0) {
+		return false
+	}
+
+	// check digest every poll interval
+	if time.Now().After(lastCheckedTimestamp.Add(interval)) &&
+		source.CreationTimestamp.Add(interval).Before(time.Now()) {
+		lastCheckedTimestamp = time.Now()
+		go c.getPodDigest(source, podDigestChan)
+	}
+
+	select {
+	case newCatalogSourceImageDigest := <-podDigestChan:
+		if newCatalogSourceImageDigest == source.Pod().Status.ContainerStatuses[0].ImageID {
+			return false
+		}
+		// we have a new imageID in our catalog source container: a new version of the catalog source image exists
+		// update catalog source pod
+		return true
+	default:
+		return false
+	}
+}
+
+// getPodDigest is an internal method that creates a pod using the latest catalog source.
+// Once the pod comes up, it puts the container image digest onto a channel.
+func (c *GrpcRegistryReconciler) getPodDigest(source grpcCatalogSourceDecorator, podChan chan string) {
+	// remove label from pod to ensure service does accidentally route traffic to the pod
+	p := source.Pod()
+	p.Labels[CatalogSourceLabelKey] = ""
+
+	pod, err := c.OpClient.KubernetesInterface().CoreV1().Pods(source.GetNamespace()).Create(p)
+	if err != nil {
+		logrus.WithField("pod", "catalog-source-pod").Warn("couldn't create new catalog source pod")
+		return
+	}
+
+	var newCatalogSourceImage string
+	// check pod to get container details
+	// we are only interested in the container image id, to see if it matches the old version
+	for i := 0; i < 10; i++ {
+		newCatalogSourceImage = pod.Status.ContainerStatuses[0].ImageID
+		if newCatalogSourceImage == "" {
+			time.Sleep(10 * time.Second)
```

Comment:
Even though this asynchronous behavior is fine, I think there is potentially a blocker + some weird behavior of go routine when is in combo with `CatalogSource` sync process in catalog operator. We should just simply skip this poll cycle and move on. Given there is a log message below to report this, we should be fine in term of keeping a record of this skipping cycle. See my [PR](https://github.com/Denton24646/operator-lifecycle-manager/pull/1/).